### ISSUE TITLE
`unroll_scan=True` fails when step dimension is `None`

### ISSUE THREAD

**Issue by craffel** [craffel/nntools#57]
cc @skaae

When the # of steps dimension of an `InputLayer` is `None`, `unroll_scan=True` fails:

``` Python
In [1]: import lasagne

In [2]: l_in = lasagne.layers.InputLayer((None, None, 2))

In [3]: l_rec = lasagne.layers.RecurrentLayer(l_in, 3, unroll_scan=True)

In [4]: lasagne.layers.get_output(l_rec)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-d11ff79b6457> in <module>()
----> 1 lasagne.layers.get_output(l_rec)

lasagne/layers/helper.py in get_output(layer_or_layers, inputs, **kwargs)
    226                                  "mapping this layer to an input expression."
    227                                  % layer)
--> 228             all_outputs[layer] = layer.get_output_for(layer_inputs, **kwargs)
    229     # return the output(s) of the requested layer(s) only
    230     try:

lasagne/layers/recurrent.py in get_output_for(self, input, mask, **kwargs)
    314                 go_backwards=self.backwards,
    315                 non_sequences=non_seqs,
--> 316                 n_steps=self.input_shape[1])[0]
    317         else:
    318             # Scan op iterates over first dimension of input and repeatedly

lasagne/utils.py in unroll_scan(fn, sequences, outputs_info, non_sequences, n_steps, go_backwards)
    337
    338         # When backwards reverse the recursion direction
--> 339         counter = range(n_steps)
    340         if go_backwards:
    341             counter = counter[::-1]

TypeError: range() integer end argument expected, got NoneType.
```

This works fine when `unroll_scan=False`:

``` Python
In [8]: l_rec = lasagne.layers.RecurrentLayer(l_in, 3, unroll_scan=False)

In [9]: lasagne.layers.get_output(l_rec)
Out[9]: DimShuffle{1,0,2}.0
```

This is expected, and I can't think of a way to fix this, so I'd suggest that we just need to make a more helpful error in this case, and make note of the fact that the # of steps dimension must not be `None` when `unroll_scan=True` in the docstring.

**Comment by skaae** [comment]
I agree. Will you fix it or should i do it?

**Comment by skaae** [comment]
I added the Error here: https://github.com/craffel/nntools/pull/58

**Comment by craffel** [comment]
Closed via #58
